test/e2e/metering_manual_install_test.go

```diff
@@ -107,6 +114,105 @@ func testManualMeteringInstall(
 	assert.NoError(t, err, "capturing logs and uninstalling metering should produce no error")
 }
 
+func createNFSProvisioner(ctx *deployframework.DeployerCtx) error {
+	//Waiting for Metering pods
+	ctx.TargetPodsCount = nonHDFSTargetPodCount
+
+	_, err := createTestingNamespace(ctx.Client, "nfsTestingNamespace")
+	if err != nil {
+		return err
+	}
+
+	files := map[string]string{
+		"pod":          "/test/e2e/manifest/nfs/nfs-server.yaml",
+		"service":      "/test/e2e/manifest/nfs/nfs-service.yaml",
+		"storageClass": "/test/e2e/manifest/nfs/nfs-storageclass.yaml",
+	}
+	server := &corev1.Pod{}
+	service := &corev1.Service{}
+	storageClass := &storagev1beta1.StorageClass{}
+	for name, file := range files {
+
+		switch name {
+		case "pod":
+			if err := deploy.DecodeYAMLManifestToObject(filepath.Join(ctx.nfsm, file), server)); err != nil {
+				return err
+			}
+			ctx.Client.CoreV1().Pods("nfsTestingNamespace").Create(context.TODO(), server, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		case "service":
+			if err := deploy.DecodeYAMLManifestToObject(file, service); err != nil {
+				return err
+			}
+			ctx.Client.CoreV1().Services("nfsTestingNamespace").Create(context.TODO(), service, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		case "storageClass":
+			if err := deploy.DecodeYAMLManifestToObject(file, storageClass); err != nil {
+				return err
+			}
+			ctx.Client.StorageV1beta1().StorageClasses().Create(context.TODO(), storageClass, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		}
+	}
+	// wait and Poll , get service
+	svc, err := ctx.Client.CoreV1().Services("nfsTestingNamespace").Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+	if apierrors.IsNotFound(err) && svc.Spec.ClusterIP == "" {
+
+		err = wait.Poll(3*time.Second, 5*time.Minute, func() (bool, error) {
+
+			svc, err = ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+
+			return svc.Spec.ClusterIP != "", err
+
+		})
+
+	}
+
+	nfsFile := "/test/e2e/manifest/nfs/nfs-persistentvolume.yaml"
+	nfs_pv := &corev1.PersistentVolume{}
```

Comment:
We shouldn't use snake casing for defining variables in Golang. Should always be camel cased: `nfsPV` is the direct substitution.